tuf/data/keys.go

```diff
@@ -317,113 +327,226 @@ func NewED25519PublicKey(public []byte) *ED25519PublicKey {
 }
 
 // Private key types
+type privateKey struct {
+	private []byte
+}
+
+type signer struct {
+	Signer crypto.Signer
```

Comment:
With the `signer` type being unexported, it already makes the `crypto.Signer` effectively unexported on the concrete private key types, but for clarity your suggestion makes a lot of sense. Update incoming.